File: 287-Find-the-Duplicate-Number/ts/main.ts
function findDuplicate(nums: number[]): number {
    let hasSeenNumber: number[] = new Array(nums.length).fill(0);
    for (let num of nums) {
        if (hasSeenNumber[num] === -1) {
            return num;
        }
        hasSeenNumber[num] = -1;
    }
    return -1;
}

File: 287-Find-the-Duplicate-Number/go/main.go
package main

import "fmt"

func main(){
	fmt.Println("hello world")
}

func findDuplicate(nums []int) int {
    hasSeenNumber := make([]int, len(nums))
    for _, num := range nums {
        if hasSeenNumber[num] == -1 {
            return num
        }
        hasSeenNumber[num] = -1
    }
    return -1
}


